packages/front-end/components/Settings/EditDataSource/DataSourceInlineEditIdentityJoins/AddEditIdentityJoinModal.tsx

```diff
@@ -102,61 +106,77 @@ export const AddEditIdentityJoinModal: FC<AddEditIdentityJoinModalProps> = ({
   }
 
   return (
-    <Modal
-      open={true}
-      submit={handleSubmit}
-      close={onCancel}
-      size="max"
-      header={modalTitle}
-      cta="Save"
-      // @ts-expect-error TS(2322) If you come across this, please fix it!: Type 'boolean | ""' is not assignable to type 'boo... Remove this comment to see the full error message
-      ctaEnabled={saveEnabled}
-      autoFocusSelector="#id-modal-identify-joins-heading"
-    >
-      <h4 id="id-modal-identify-joins-heading">Identifier Join</h4>
-      <p>Queries that return a mapping between different identifier types</p>
-
-      <div className="row">
-        <div className="col-xs-12 col-md-6">
-          <MultiSelectField
-            label="Identifier Types"
-            value={userEnteredIdentityJoinIds}
-            onChange={(value) => {
-              form.setValue("ids", value);
-            }}
-            options={identityTypes.map((idType) => ({
-              value: idType.userIdType,
-              label: idType.userIdType,
-            }))}
-          />
-
-          <div>
-            {userEnteredIdentityJoinIds.length ? (
-              <>
-                <div className="pt-md-4">
-                  <strong>Required columns</strong>
-                </div>
-                <ul>
-                  {userEnteredIdentityJoinIds.map((id) => (
-                    <li key={id}>
-                      <code>{id}</code>
-                    </li>
-                  ))}
-                </ul>
-              </>
-            ) : null}
+    <>
+      {sqlOpen && (
+        <EditSqlModal
+          close={() => setSqlOpen(false)}
+          datasourceId={dataSource.id}
+          requiredColumns={new Set(userEnteredIdentityJoinIds)}
+          value={userEnteredQuery}
+          save={async (sql) => form.setValue("query", sql)}
+        />
+      )}
+      <Modal
+        open={true}
+        submit={handleSubmit}
+        close={onCancel}
+        size="max"
+        header={modalTitle}
+        cta="Save"
+        // @ts-expect-error TS(2322) If you come across this, please fix it!: Type 'boolean | ""' is not assignable to type 'boo... Remove this comment to see the full error message
+        ctaEnabled={saveEnabled}
+        autoFocusSelector="#id-modal-identify-joins-heading"
+      >
+        <h4 id="id-modal-identify-joins-heading">Identifier Join</h4>
+        <p>Queries that return a mapping between different identifier types</p>
+        <div>{form.watch("query")}</div>
```

Comment:
Remove this div.  The query is already shown below in the `<Code>` component

Reply:
Done, it is removed now